content/learning-paths/smartphones-and-mobile/function-multiversioning/implementation-details.md

```diff
@@ -0,0 +1,42 @@
+---
+title: Implementation Details
+weight: 4
+
+### FIXED, DO NOT MODIFY
+layout: learningpathall
+---
+
+## Implementation details
+
+In order to select the most appropriate version of a function, each call to a versioned function is routed through an indirect function resolver which is pointed by the called symbol (IFUNC). The compiler generates a resolver based on the function versions declared in the translation unit. A typical resolver implementation uses a runtime library to detect the presence of the architectural features on which the function versions depend and returns a pointer to the right version. The resolution of the called symbol is delayed until runtime, when the dynamic loader runs the resolver and updates the procedure linkage table (PLT) with a pointer to the chosen implementation. The resolver function is run only once and its returned value remains unchanged for the lifetime of the process. References to the called symbol are handled by relocations which return the cached PLT entry.
+
+#### Differences between GCC 14 and LLVM 19 implementations
+
+- The attribute `target_version` in GCC is only supported for C++, not for C.
```

Comment:
Do you also want to mention `target_clones` here?

Reply:
No, target_clones is supported for C as well in gcc-14.